test/GeneratorsTests.cpp

```diff
@@ -0,0 +1,64 @@
+#include "catch.hpp"
+#include "UtilityFunctions.hpp"
+#include "FibonacciNumberGenerator.hpp"
+#include "PrimeNumberGenerator.hpp"
+
+
+SCENARIO("Functional tests")
+{
+    GIVEN("One Fibonacci generator and numberOfValues = 10")
+    {
+        std::vector<GeneratorType> collection = {};
+        auto numberOfValues = 10;
+        collection.push_back(GeneratorType::FIBONACCI);
+
+        WHEN("sumNumbersFromGenerators is called")
+        {
+            auto result = sumNumbersFromGenerators(collection, numberOfValues);
+
+            THEN("It should return same value as sumGeneratedNumbers")
+            {
+                CHECK(result == sumGeneratedNumbers(std::move(std::make_unique<FibonacciNumberGenerator>()), numberOfValues));
```

Comment:
`std::move()` is redundant here. The code will work in the same way without it. The same for other places.

Reply:
The compiler also complains that it prevents optimisation: https://travis-ci.com/coders-school/pre_work/jobs/260930593

Reply:
True :) corrected